### PR TITLE
chore: use Towncrier for Changelog management

### PR DESCRIPTION
**Pull request by astencel-sumo**

### CODE REVIEW

File: .changelog/changelog_template.jinja
Comment (sumo-drosiek) [inline .changelog/changelog_template.jinja:1]: Could you add links to upstream releases as well?

Comment (astencel-sumo) [reply .changelog/changelog_template.jinja:1]: Hm we actually never included the upstream release links in the Changelog. Can we make it a separate PR?

Comment (sumo-drosiek) [reply .changelog/changelog_template.jinja:1]: Oh, it's changelog. We may skip it then

**Review by sumo-drosiek — APPROVED** [review]

**Review by swiatekm-sumo — APPROVED** [review]
We don't actually need to use prettier here. In https://github.com/SumoLogic/sumologic-kubernetes-collection we're using it to format MD and yaml files anyway, so it makes sense, but in this repo we don't have a formatter for markdown at all, so maybe it's not worth it?

**Comment by astencel-sumo** [comment]
> We don't actually need to use prettier here. In https://github.com/SumoLogic/sumologic-kubernetes-collection we're using it to format MD and yaml files anyway, so it makes sense, but in this repo we don't have a formatter for markdown at all, so maybe it's not worth it?

I only added it because Towncrier insists on adding two newlines at the end of its Changelog entry, which needs to be replaced with a single newline for the Mardown linter to pass. Prettier is the easiest way to do this.

**Pull request merged**